File: src/main/scala/leetcode/com/MoveZeroes.scala
package leetcode.com

object MoveZeroes {
  def moveZeroes(nums: Array[Int]): Unit = {
    var j: Int = -1
    var tmp: Int = 0
    for (i <- 0 until nums.length) {

      if (nums(i) != 0) {
        j += 1;
        tmp = nums(i)
        nums(i) = nums(j)
        nums(j) = tmp
      }
    }
  }
}

File: GoProject/leetcode/array/move-zeroes.go
package array

func moveZeroes(nums []int) {
	slow, quick := 0, 1

	l := len(nums) - 1

	for i := 0; i < l; i++ {
		switch {
		case nums[slow] == 0 && nums[quick] != 0:
			nums[slow], nums[quick] = nums[quick], nums[slow]
			slow += 1
			quick += 1
		case nums[slow] != 0 && nums[quick] == 0:
			slow += 1
			quick += 1
		case nums[slow] == 0 && nums[quick] == 0:
			quick += 1
		case nums[slow] != 0 && nums[quick] != 0:
			slow += 1
			if slow == quick {
				quick += 1
			}
		}
	}

}
